enhancements/cluster-logging/cluster-logging-move-init-to-es-operator.md

```diff
@@ -0,0 +1,65 @@
+---
+title: cluster-logging-log-forwarding
+authors:
+  - "@jcantrill"
+  - "@igor-karpukhin"
+  - "@ewolinetz"
+reviewers:
+  - "@bparees"
+  - "@jeremyeder"
+  - "@jcantrill"
+  - "@igor-karpukhin"
+  - "@ewolinetz"
+approvers:
+  - "@bparees"
+  - "@ewolinetz"
+  - "@richm"
+creation-date: 2019-09-17
+last-updated: 2019-10-25
+status: implementable
+see-also:[]
+replaces:[]
+superseded-by:[]
+---
+
+# cluster-logging-es-init
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The purpose of moving Elasticsearch initialization logic to the elasticsearch-operator is to reduce the number of times these init scripts executed.
+At the moment we keep these scripts in the ES image, and execute them everytime when ES pod starts. 
+
+
+### Goals
+The specific goals of this proposal are:
+
+* Provide an API that allows users to execute init procedures for the ES cluster
+* Allow users to choose what command has to be executed during the init process for FULL and ROLLING elasticsearch cluster restart
+
+## Proposal
+
+The following elasticsearch-operator changes will allow to execute init process for ES cluster only when we restarting the whole cluster, not it's pods:
+
+
+* Adding additional section to the elasticsearch-operator called `bootstrap`, which may look like this:
+
+    ```json
+    "bootstrap": {
+      "image": "quay.io/openshift/origin-logging-elasticsearch-bootstrap6:latest",
+      "actions: {
+          "onFullRestart": "init.sh"
+          "onRollingRestart": "init_rolling.sh"
+      }
+    },
+    ```
+* Moving all the init scripts from `origin-aggregated-logging` into the `elasticseach-operator`
```

Comment:
There really isn't a need for us specifically to move the scripts per se; we can leave them in our Elasticsearch image and use an alternate entrypoint, and alter the existing run image to do not do init

Reply:
Fixed. We then will build an image in the `origin-aggregated-logging` repo

Reply:
Fixed. We then will build an image in the `origin-aggregated-logging` repo